File: two pointers/lc485.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    nlen = nums.length
    //base cases
    if(nlen == 1){
        if(nums[0] == 1){
            return 1
        }
        if(nums[0] == 0){
            return 0
        }
    }

    //define pointers
    let lp = 0
    let rp = 0
    let max = 0

    while (rp < nlen){
        if (nums[rp] == 0){
            rp++
            lp = rp
        } else {
            max = Math.max(max, (rp-lp+1))
            rp++
        }
    }

    return max


};

File: two pointers/lc485.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        nlen = len(nums)

        if nlen == 1:
            if nums[0] == 0:
                return 0
            if nums[0] ==1:
                return 1

        lp = 0
        rp = 0
        maxL = 0
        
        while rp < nlen:
            if nums[rp] == 0:
                rp+=1
                lp = rp
            else:
                maxL = max(maxL, (rp-lp+1))
                rp+=1

        return maxL
